pkg/client/restclient/request.go

```diff
@@ -811,7 +811,18 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			return err
+			if !net.IsConnectionReset(err) {
+				return err
+			}
+			// "Connection reset by peer" is usually a transient error.
```

Comment:
Removing label while we discuss.

We can get this error anytime from the moment the connection is opened until the moment the server flushes the headers.  So no, it's not safe for anything where the client doesn't expect idempotence.

This was just a statistical reliability fix, right?  If so, I'd expect GET to be sufficient.